dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/export/datacite/DataCiteResourceCreator.java

```diff
@@ -130,6 +135,42 @@ private List<RelatedIdentifier> extractRelatedIdentifiers(DvObject dvObject) {
         return Collections.emptyList();
     }
 
+    private List<RelatedIdentifier> extractAdditionalRelatedIdentifiers(Dataset dataset) {
+        List<RelatedIdentifier> identifiers = new ArrayList<>();
+        DatasetVersion version = dataset.getLatestVersion();
+        List<Map<String, DatasetField>> relatedElements = version.extractFieldsWithSubfields(DatasetFieldConstant.publication,
+                DatasetFieldConstant.relatedDataset, DatasetFieldConstant.relatedMaterial);
+        for (Map<String, DatasetField> fieldMap : relatedElements) {
+            String parentName = extractParentFieldName(fieldMap);
+            String[] fields = RelatedIdFields.getSuffixes()
+                    .map(parentName::concat)
+                    .map(fieldMap::get)
+                    .map(f -> f != null ? f.getValue() : null)
+                    .toArray(String[]::new);
+            String relatedIdType = fields[RelatedIdFields.RELATED_ID_TYPE.ordinal()];
+            if (StringUtils.isBlank(relatedIdType)) {
+                continue;
+            }
+            Map<String, String> alternativeToMainIdTypeIndex = RelatedIdentifierTypeConstants.getAlternativeToMainIdTypeIndex();
+            if (alternativeToMainIdTypeIndex.containsKey(relatedIdType)) {
+                relatedIdType = alternativeToMainIdTypeIndex.get(relatedIdType);
+            }
+            identifiers.add(new RelatedIdentifier(
+                    StringUtils.containsIgnoreCase(relatedIdType, "url")
+                            ? fields[RelatedIdFields.URL.ordinal()]
+                            : fields[RelatedIdFields.RELATED_ID.ordinal()],
+                    fields[RelatedIdFields.RELATION_TYPE.ordinal()], relatedIdType));
```

Comment:
In schema relation type is required field (view-source:http://schema.datacite.org/meta/kernel-4/metadata.xsd):
```
              <xs:element name="relatedIdentifier" minOccurs="0" maxOccurs="unbounded">
                <xs:annotation>
                  <xs:documentation>Identifiers of related resources. Use this property to indicate subsets of properties, as appropriate.</xs:documentation>
                </xs:annotation>
                <xs:complexType>
                  <xs:simpleContent>
                    <xs:extension base="xs:string">
                      <xs:attribute name="resourceTypeGeneral" type="resourceType" use="optional" />
                      <xs:attribute name="relatedIdentifierType" type="relatedIdentifierType" use="required" />
                      <xs:attribute name="relationType" type="relationType" use="required" />
                      <xs:attribute name="relatedMetadataScheme" use="optional" />
                      <xs:attribute name="schemeURI" type="xs:anyURI" use="optional" />
                      <xs:attribute name="schemeType" use="optional" />
                    </xs:extension>
                  </xs:simpleContent>
                </xs:complexType>
              </xs:element>
```

Reply:
Ok – the cases with empty `relationType` will be filtered out.